modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -133,20 +141,28 @@ private void populateSource(Source source, OMElement sourceEle) {
     private void populateTarget(Target target, OMElement sourceEle) {
         // type attribute
         OMAttribute typeAttr = sourceEle.getAttribute(ATT_TYPE);
+        OMAttribute actionAttr = sourceEle.getAttribute(ATT_ACTION);
+
+        if (actionAttr != null && actionAttr.getAttributeValue() != null) {
+            target.setAction(actionAttr.getAttributeValue());
+        } else {
+            target.setAction("replace");
+        }
+
         if (typeAttr != null && typeAttr.getAttributeValue() != null) {
-            int type = convertTypeToInit(typeAttr.getAttributeValue());
+            int type = convertTypeToInt(typeAttr.getAttributeValue());
             if (type >= 0) {
                 target.setTargetType(type);
+                if (type == 1) {
+                    if (!target.getAction().equals("replace")) {
```

Comment:
Let's merge two if conditions to one

And also please use EnrichMediator.BODY etc instead of using integer values for comparison

Reply:
For `if (type >= 0)` these kind of checking I need integers. So as used in old implementation I use integers here.